Source/Scene/Cesium3DTileGroundPrimitive.js

```diff
@@ -0,0 +1,875 @@
+                });
+            }
+        }
+
+        primitive._batchedIndices = batchedDrawCalls;
+
+        for (i = 0; i < colorsLength; ++i) {
+            color = colors[i];
+            rgba = color.toRgba();
+
+            object = buffers[rgba];
+            var positionOffset = object.offset;
+            var positionIndex = positionOffset * 3;
+            var colorIndex = positionOffset * 4;
+            var idIndex = positionOffset;
+
+            var polygonOffset = offsets[i];
+            var polygonCount = counts[i];
+
+            var minLat = Number.POSITIVE_INFINITY;
+            var maxLat = Number.NEGATIVE_INFINITY;
+            var minLon = Number.POSITIVE_INFINITY;
+            var maxLon = Number.NEGATIVE_INFINITY;
+
+            for (j = 0; j < polygonCount; ++j) {
+                var encodedPosition = Cartesian3.unpack(positions, polygonOffset * 3 + j * 3, scratchEncodedPosition);
+                var rtcPosition = Matrix4.multiplyByPoint(decodeMatrix, encodedPosition, encodedPosition);
+                var position = Cartesian3.add(rtcPosition, center, rtcPosition);
+
+                var carto = ellipsoid.cartesianToCartographic(position, scratchBVCartographic);
+                var lat = carto.latitude;
+                var lon = carto.longitude;
+
+                minLat = Math.min(lat, minLat);
+                maxLat = Math.max(lat, maxLat);
+                minLon = Math.min(lon, minLon);
+                maxLon = Math.max(lon, maxLon);
+
+                var normal = ellipsoid.geodeticSurfaceNormal(position, scratchNormal);
+                var scaledPosition = ellipsoid.scaleToGeodeticSurface(position, position);
+                var scaledNormal = Cartesian3.multiplyByScalar(normal, minHeight, scratchScaledNormal);
+                var minHeightPosition = Cartesian3.add(scaledPosition, scaledNormal, scratchMinHeightPosition);
+
+                scaledNormal = Cartesian3.multiplyByScalar(normal, maxHeight, scaledNormal);
+                var maxHeightPosition = Cartesian3.add(scaledPosition, scaledNormal, scratchMaxHeightPosition);
+
+                Cartesian3.subtract(maxHeightPosition, center, maxHeightPosition);
+                Cartesian3.subtract(minHeightPosition, center, minHeightPosition);
+
+                Cartesian3.pack(maxHeightPosition, batchedPositions, positionIndex);
+                Cartesian3.pack(minHeightPosition, batchedPositions, positionIndex + 3);
+
+                batchedIds[idIndex] = i;
+                batchedIds[idIndex + 1] = i;
+
+                positionIndex += 6;
+                colorIndex += 8;
+                idIndex += 2;
+            }
+
+            var rectangle = scratchBVRectangle;
+            rectangle.west = minLon;
+            rectangle.east = maxLon;
+            rectangle.south = minLat;
+            rectangle.north = maxLat;
+
+            boundingVolumes[i] = OrientedBoundingBox.fromRectangle(rectangle, minHeight, maxHeight, ellipsoid);
```

Comment:
Couldn't we compute much better bounding volumes using the Cartesian positions?  For example, if the polygon is thin like a polyline with a bearing angle of 45 degrees, this would be a box around the diangle instead of a tight fit OBB.  What is the concern?  Performance of computing the tight OBB?